src/main/java/org/broadinstitute/dsde/consent/ontology/datause/DataUseMatchCasesV3.java

```diff
@@ -0,0 +1,273 @@
+    boolean purposeHMB = getNullableOrFalse(purpose.getHmbResearch());
+    boolean datasetGRU = getNullableOrFalse(dataset.getGeneralUse());
+    boolean datasetHMB = getNullableOrFalse(dataset.getHmbResearch());
+
+    // Deny purpose GRU condition
+    if (datasetHMB && purposeGRU) {
+      return MatchResult.from(MatchResultType.DENY, Collections.singletonList(HMB_F1));
+    }
+
+    // Approve dataset GRU condition
+    if (datasetGRU) {
+      return MatchResult.from(MatchResultType.APPROVE, Collections.emptyList());
+    }
+
+    // Approve dataset HMB condition
+    if (purposeHMB && datasetHMB) {
+      return MatchResult.from(MatchResultType.APPROVE, Collections.emptyList());
+    } else {
+      return MatchResult.from(MatchResultType.DENY, Collections.singletonList(HMB_F2));
+    }
+  }
+
+  /**
+   * RP: Study population origins or ancestry
+   * Future use is limited to research involving a specific population [POA]
+   * Datasets:
+   *      Any dataset tagged with GRU
+   *      Any dataset tagged with POA
+   */
+  static MatchResult matchPOA(DataUseV3 purpose, DataUseV3 dataset) {
+    // short-circuit if no POA clause
+    if (Objects.isNull(purpose.getPopulationOriginsAncestry())) {
+      return MatchResult.from(MatchResultType.APPROVE, Collections.emptyList());
+    }
+
+    boolean purposePOA = getNullableOrFalse(purpose.getPopulationOriginsAncestry());
+    boolean datasetPOA = getNullableOrFalse(dataset.getPopulationOriginsAncestry());
+    boolean datasetGRU = getNullableOrFalse(dataset.getGeneralUse());
+
+    // Approve dataset GRU condition
+    if (datasetGRU) {
+      return MatchResult.from(MatchResultType.APPROVE, Collections.emptyList());
+    }
+
+    // Approve dataset POA condition
+    if (purposePOA && datasetPOA) {
+      return MatchResult.from(MatchResultType.APPROVE, Collections.emptyList());
+    }
+
+    List<String> failures = new ArrayList<>();
+    if (!(purpose.getPopulationOriginsAncestry() &&
+        getNullableOrFalse(dataset.getGeneralUse()))) {
+      failures.add(POA_F1);
+    }
+
+    return MatchResult.from(MatchResultType.DENY, failures);
+  }
+
+  /**
+   * RP: Methods development/Validation study
+   * Future use for methods research (analytic/software/technology development) outside the
+   *      bounds of the other specified restrictions
+   * Datasets:
+   *      Any dataset where GRU = true
+   *      Any dataset where HMB = true
+   *      Any dataset where POA = true
+   *      Any dataset where DS-X match
+   */
+  static MatchResult matchMDS(DataUseV3 purpose, DataUseV3 dataset, MatchResultType diseaseMatch) {
+    // short-circuit if no disease focused research
+    if (purpose.getDiseaseRestrictions().isEmpty() && dataset.getDiseaseRestrictions().isEmpty()) {
+      return MatchResult.from(MatchResultType.APPROVE, Collections.emptyList());
+    }
+
+    boolean purposeMDS = getNullableOrFalse(purpose.getMethodsResearch());
```

Comment:
I think there might be a logic error here. There are technically three cases, but I think null or false are the same as you've calculated here.

1. Purpose MDS is true - we need to check the cases as outlined in the doc.
2. Purpose MDS is false/null - we don't need to do any checks - we can short circuit this because it isn't relevant to any decision making.